### PR TITLE
Add Docker Samples

### PR DESCRIPTION
**Pull request by marrobi**
We recently had the need to run FHIR Server locally on a developers machine. 

This PR contains the resulting Dockerfile, and and sample Docker Compose and Kubernetes configurations. Samples use a SQL Server container for the back end storage, although PaaS database services could also be used.

Content is split across two sample directories with corresponding `readme.md` files.

This has been tested on WSLv2 with Docker Compose and Kubernetes on Azure Kubernetes Service. I have written 1,000,000+ resources and performed searches.

Welcome feedback around the PR, especially the Dockerfile and the way the project is being built.

### CODE REVIEW

**Comment by marrobi** [comment]
@hansenms  not sure why this failed as the global.json file in the root of the repos is the original. I made this change in https://github.com/microsoft/fhir-server/pull/754/commits/7fe452f60d5675640b480a0f53c7eb0da0c409b4 .

**Comment by hansenms** [comment]
It must be picking up the other global.json too. Can you rename it?

**Comment by marrobi** [comment]
@hansenms want to try now?

**Comment by hansenms** [comment]
@marrobi build is working now, but cred scan is failing. You have some passwords in there. It will not work.

**Comment by marrobi** [comment]
I've removed a couple of example passwords - is there any way to tell whats causing the build to fail? Looks like it gets deleted on the server?

File: samples/kubernetes/README.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+Create a Kubernetes secret to store the SQL Server SA password, `<SA_PASSWORD>`:
+
+```bash
+kubectl create secret generic sql-settings --from-literal=password='<SA_PASSWORD>'
+```
+
+### Deploy SQL Server
+
+The example SQL server deployment creates a Kubernetes deployment, load balancer of type `ClusterIP`, and a persistant volume claim is created with aa storage class of `volume.beta.kubernetes.io/storage-class: managed-premium`. If you are not running Azure Kubernetes Service you will need to adjust the storage class to one that exists on your cluster.
```
Comment (hansenms) [inline samples/kubernetes/README.md]: nit: "aa storage class..." -> "a storage class"

File: samples/kubernetes/README.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
+# Running in Kubernetes
+
+*IMPORTANT:* This configuration is not suitable for production scenarios. This sample is provided as an example of how the Fhir Server project can be run on Kubernetes. Please note by default the SQL server connection is not encrypted and authentication on the FHIR Server has been disabled.
+
+If you do not have a Kubernetes cluster and want to deploy Kubernetes on Azure documentation can be found here: [https://docs.microsoft.com/en-us/azure/aks/](https://docs.microsoft.com/en-us/azure/aks/)
+
+## Optional: Deploying SQL Server on Kubernetes
+
+The FHIR Server can be connected to a SQL server intance external to Kubernetes, for example Azure SQL Database. However if you wish to run SQL Server on Kubernetes follow the steps below:
```
Comment (hansenms) [inline samples/kubernetes/README.md]: nit: intance -> instance

**Comment by hansenms** [comment]
@brendankowitz can you take a look at this. It could be a lot more elaborate, but it is a good start. Should we break this into separate PRs for Dockerfile, compose, Kubernetes?

File: samples/docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+COPY ./src/ ./src/
+COPY ./tools/Microsoft.Health.Extensions.BuildTimeCodeGenerator/ ./tools/Microsoft.Health.Extensions.BuildTimeCodeGenerator/
+
+RUN dotnet add "./tools/Microsoft.Health.Extensions.BuildTimeCodeGenerator/Microsoft.Health.Extensions.BuildTimeCodeGenerator.csproj" package Microsoft.CodeAnalysis.Analyzers  --version 2.9.4
+
+RUN dotnet publish "./src/Microsoft.Health.Fhir.${FHIR_VERSION}.Web/Microsoft.Health.Fhir.${FHIR_VERSION}.Web.csproj" -c Release -o "/build"
+
+FROM mcr.microsoft.com/dotnet/core/aspnet:2.2-alpine AS runtime
+
+ARG FHIR_VERSION=Stu3
```
[the remark this answers is not in the document]
Comment (marrobi) [reply samples/docker/Dockerfile]: Changed default, and taken out k8s - will work on separate PR with helm chart.

File: samples/kubernetes/README.md (hunk 2 above)
Comment (marrobi) [reply samples/kubernetes/README.md]: Taken k8s out. Will fix in new PR.

File: samples/kubernetes/README.md (hunk 1 above)
Comment (marrobi) [reply samples/kubernetes/README.md]: Taken k8s out. Will fix in new PR.

**Comment by hansenms** [comment]
@johnstairs

**Comment by brendankowitz** [comment]
Thanks @marrobi, I think this looks like a good starting point. I'd like for us to get a build going for the docker container too, I don't think I'd hold this PR up on that though

**Comment by hansenms** [comment]
@brendankowitz do you want to approve this and we could merge it in as a starting point and then work on setting up a container build.

**Comment by brendankowitz** [comment]
Adding story #779

**Pull request merged**